src/widgets/block.rs

```diff
@@ -396,16 +400,36 @@ impl<'a> Block<'a> {
         self
     }
 
-    /// Defines the block style.
+    /// Defines the style of the entire block.
     ///
     /// This is the most generic [`Style`] a block can receive, it will be merged with any other
-    /// more specific style. Elements can be styled further with [`Block::title_style`] and
-    /// [`Block::border_style`].
+    /// more specific styles. Elements can be styled further with [`Block::title_style`] and
+    /// [`Block::border_style`], which will be applied on top of this style. If the block is used
+    /// as a container for another widget, the inner widget can also be styled.
     ///
     /// `style` accepts any type that is convertible to [`Style`] (e.g. [`Style`], [`Color`], or
     /// your own type that implements [`Into<Style>`]).
     ///
     /// This will also apply to the widget inside that block, unless the inner widget is styled.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// let block = Block::new().style(Style::new().red().on_black());
+    ///
+    /// // To override styles set on the block level, you can style the title, border.
+    /// let block = Block::new()
+    ///     .style(Style::new().red().bold().italic())
+    ///     .border_style(Style::new().not_italic()) // will be red and bold
+    ///     .title_style(Style::new().not_bold()) // will be red and italic
+    ///     .title("Title");
```

Comment:
I think the sentence in from of this should be phrased more clearly, but I don't have a good idea how yet.
Currently, it reads like `border_style` overrides, but actually the `.not_italic()` overrides.

Reply:
Funnily, the user that was originally confused about this implementation stated they understood the explanation well. 😀.

The styles are actually overridden as specified, style by style, not element by element.

Reply:
Ah I can see how what I wrote is possibly confusing. The way that one style overrides another is by combination of the parts that are specified overwriting those that are previously specified. That's the "overriding" that I meant here.

This is called our more explicitly in [`Style`](https://docs.rs/ratatui/0.26.3/ratatui/style/struct.Style.html):

> Styles represents an incremental change. If you apply the styles S1, S2, S3 to a cell of the terminal buffer, the style of this cell will be the result of the merge of S1, S2 and S3, not just S3.

Reply:
Added comment in 1e7b6f47a